app/models/company.rb

```diff
@@ -236,8 +239,44 @@ def featured_competitors
     end
   end
 
+  def create_snapshot_doc!
+    # Download template doc
+    content = google_drive.export(team.snapshot_template, 'text/html').string
+
+    # Fill in fields
+    content.gsub! '{COMPANY_NAME}', name
+    content.gsub! '{TRELLO_URL}', link_to(card.trello_url, card.trello_url)
+    content.gsub! '{PARTNER_NAMES}', users.map(&:cached_name).join(', ')
+
+    # Extract body (avoids style problems)
+    content = Nokogiri::HTML(content).at('body').inner_html
+
+    # Create new doc
+    snapshot = google_drive.create(
+      "#{name.gsub(/['"]/, '')} Snapshot",
+      'application/vnd.google-apps.document',
+      StringIO.new(content),
+      team.snapshot_folder_ids.last,
+      'text/html'
+    ).web_view_link
+
+    # Update pitch if it exists
+    # Otherwise, this snapshot will get associated by name when the pitch is created
+    if pitch.present?
+      pitch.snapshot = snapshot
+      pitch.save!
+    end
+
+    snapshot
+  end
+
   private
 
+  def google_drive
```

Comment:
We have this same code somewhere else, maybe Pitch? Let's refactor it into a concern.